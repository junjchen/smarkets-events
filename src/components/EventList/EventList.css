.event-list__title {
    font-weight: 400;
    text-transform: uppercase;
}

.event-list__status {
    font-size: 0.8em;
    padding: 0.5em;
    color: rgb(255, 255, 255);
    background-color: rgb(67, 160, 71);
}

.event-list__status p {
    margin: 0.5em 0;
}

.event-list__status--error {
    background-color: rgb(244, 81, 30);
}

.event-list__status--fetching {
    background-color: rgb(117, 117, 117);
}