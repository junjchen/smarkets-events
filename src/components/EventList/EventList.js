import React, {Component} from 'react'
import EventCard from '../EventCard'
import StatusBar from '../StatusBar'
import './EventList.css'

class EventList extends Component {
    componentDidMount() {
        const {fetchData} = this.props
        fetchData()
    }
    render() {
        const {title, events, selectEvent} = this.props
        return (
            <div className="event-list">
                <StatusBar {...this.props}/>
                <h1 className="event-list__title">{title}</h1>
                {events.length > 0 && (
                    <div className="event-list__container">
                        {events && events.map((x, i) => <EventCard key={i} onSelect={selectEvent} {...x}/>)}
                    </div>
                )}
            </div>
        )
    }
}

export default EventList