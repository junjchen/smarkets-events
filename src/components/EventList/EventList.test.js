import React from 'react'
import {shallow} from 'enzyme'
import EventList from './EventList'

const createProps = () => ({
    events: [
        {
            id: 1,
            name: 'name',
            short_name: 'short_name',
            parent_name: 'category',
            start_datetime: new Date('1990/10/10')
        }, {
            id: 2,
            name: 'name',
            short_name: 'short_name',
            parent_name: 'category',
            start_datetime: new Date('1990/10/10')
        }
    ],
    fetchData: jest.fn()
})

describe('EventList', () => {
    it('renders same as snapshot', () => {
        const props = createProps()
        const wrapper = shallow(<EventList {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })
    it('calls fetchData', () => {
        const props = createProps()
        const wrapper = shallow(<EventList {...props}/>)
        expect(props.fetchData).toBeCalled()
    })
})
