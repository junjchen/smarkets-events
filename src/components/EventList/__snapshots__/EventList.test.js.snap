// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList renders same as snapshot 1`] = `
<div
  className="event-list"
>
  <StatusBar
    events={
      Array [
        Object {
          "id": 1,
          "name": "name",
          "parent_name": "category",
          "short_name": "short_name",
          "start_datetime": 1990-10-09T23:00:00.000Z,
        },
        Object {
          "id": 2,
          "name": "name",
          "parent_name": "category",
          "short_name": "short_name",
          "start_datetime": 1990-10-09T23:00:00.000Z,
        },
      ]
    }
    fetchData={[Function]}
  />
  <h1
    className="event-list__title"
  />
  <div
    className="event-list__container"
  >
    <EventCard
      id={1}
      key="0"
      name="name"
      parent_name="category"
      short_name="short_name"
      start_datetime={1990-10-09T23:00:00.000Z}
    />
    <EventCard
      id={2}
      key="1"
      name="name"
      parent_name="category"
      short_name="short_name"
      start_datetime={1990-10-09T23:00:00.000Z}
    />
  </div>
</div>
`;
